packages/blockchain-tools/src/contracts/soroban/credits/src/tests/utils.rs

```diff
@@ -0,0 +1,306 @@
+
+  let carbonSac = mintable_token::Client::new(
+    &e,
+    &e.register_stellar_asset_contract_v2(admin.clone()).address()
+  );
+
+  (xlm, usdc, carbonSac)
+}
+
+pub fn create_credit_contract<'a>(
+  e: &Env,
+  admin: &Address,
+  initiative: &String,
+  provider: &Address,
+  vendor: &Address,
+  bucket: i128,
+  xlm: &Address,
+  usdc: &Address,
+  carbonSac: &Address,
+  sink: &Address,
+  soroswapRouter: &Address
+) -> CreditsClient<'a> {
+  info!("Creating credit contract...");
+
+  let contract_id = e.register(
+    Credits,
+    (
+      admin.clone(),
+      initiative.clone(),
+      provider.clone(),
+      vendor.clone(),
+      bucket,
+      xlm.clone(),
+      usdc.clone(),
+      carbonSac.clone(),
+      sink.clone(),
+      soroswapRouter.clone()
+    )
+  );
+  let contract_client = CreditsClient::new(e, &contract_id);
+  info!("Credit Contract created!");
+  contract_client
+}
+
+pub fn create_sink_successors(e: &Env) -> (Address, Address) {
+  let admin = Address::generate(e);
+  let carbon_sac = Address::generate(e);
+  let carbonsink_sac = Address::generate(e);
+
+  let first_sink_id = e.register(
+    sink_contract::WASM, 
+    (&admin, &carbon_sac, &carbonsink_sac)
+  );
+  let first_sink_client = sink_contract::Client::new(e, &first_sink_id);
+  let second_sink_id = e.register(
+    sink_contract::WASM, 
+    (&admin, &carbon_sac, &carbonsink_sac)
+  );
+  first_sink_client.set_contract_successor(&second_sink_id);
+
+  let second_sink_client = sink_contract::Client::new(e, &first_sink_id);
+  let third_sink_id = e.register(
+    sink_contract::WASM, 
+    (&admin, &carbon_sac, &carbonsink_sac)
+  );
+  second_sink_client.set_contract_successor(&third_sink_id);
+
+  (first_sink_id, third_sink_id)
+}
+
+pub fn setup_soroswap_env<'a>(
+  e: &Env,
+  admin: &Address,
+  xlm: &token::Client<'a>,
+  usdc: &mintable_token::Client<'a>,
+  carbonSac: &mintable_token::Client<'a>,
+) -> soroswap_router::Client<'a> {
+  let xlm_liqudity: i64 = 1_000_000_000_000_000;
+  let usdc_liquidity: i128 = 1_000_000_000_000_000;
+  let carbon_liquidity: i128 = 1_000_000_000_000_000;
+
+  let xlm_minter_pubkey = "GA2H3SJYGIUG2DXXUZ7IN3LNO2AIMVWCDCL25PKQHKMC76OWW3HYQHY4";
+  let xlm_minter_address = Address::from_str(&e, xlm_minter_pubkey);
+  create_account_entry(&e, &xlm_minter_pubkey, i64::MAX);
+
+  xlm.transfer(&xlm_minter_address, &admin, &(i64::MAX as i128));
+  usdc.mint(&admin, &(usdc_liquidity * 2));
+  carbonSac.mint(&admin, &(carbon_liquidity * 2));
+
+  // prepare soroswap factory
+  let pair_hash = e.deployer().upload_contract_wasm(soroswap_pair::WASM);
+  let factory_address = e.register(
+    soroswap_factory::WASM,
+    ()
+  );
+  let factory_client = soroswap_factory::Client::new(&e, &factory_address);
+  factory_client.initialize(&admin, &pair_hash);
+
+  // prepare two pairs
+  factory_client.create_pair(&xlm.address, &usdc.address);
+  factory_client.create_pair(&usdc.address, &carbonSac.address);
```

Comment:
This is not needed because it's automatically done by the router when we add liquidity.